crates/trigger-http/src/lib.rs

```diff
@@ -336,6 +338,22 @@ impl HttpTrigger {
                     TokioIo::new(stream),
                     service_fn(move |request| {
                         let self_ = self_.clone();
+                        let span = tracing::info_span!(
+                            "handle_http_request",
+                            "otel.kind" = "server",
+                            "http.request.method" = %request.method(),
+                            "network.peer.address" = %addr.ip(),
+                            "network.peer.port" = %addr.port(),
+                            "network.protocol.name" = "http",
+                            "url.path" = request.uri().path(),
+                            "url.query" = request.uri().query().unwrap_or(""),
+                            "url.scheme" = request.uri().scheme_str().unwrap_or(""),
+                            "client.address" = request.headers().get("x-forwarded-for").and_then(|val| val.to_str().ok()),
+                            // TODO(Caleb): Recorded later
+                            // "error.type" = Empty,
+                            // "http.response.status_code" = Empty,
+                            // "http.route" = Empty,
+                        );
```

Comment:
This 15-line code block dominates the lambda: it no longer stands out as "the service fn takes a request and handles it" but more "the service fn takes a request and does a metric butt-ton of tracing on it" - you have to read past a _lot_ of tracing fields to get to the actual behaviour.  Could this be encapsulated?

Reply:
I'm sure it can, but I'm scared to touch code that is this critical to Spin right before I merge this. I'll make this stylistic change in a fast-follow PR when I come back to add these attributes that are added later.

Reply:
If you extracted it into a function then the call site information would be wrong. You could extract it into a macro instead if you want.

Reply:
As a further tangent, I just learned that you can use constants as the field names, such as those from [`opentelemetry-semantic-conventions`](https://docs.rs/opentelemetry-semantic-conventions/latest/opentelemetry_semantic_conventions/trace/index.html); `tracing` just requires that you wrap constants in `{}`, e.g.:

```rust
mod keep_all_these_constants_hidden_away {
  use opentelemetry_semantic_conventions::trace::*;
  ...
    span!(
      ...
      { HTTP_REQUEST_METHOD } = method,
    )
}
```

Reply:
(but agreed this can come in a followup, if at all)